homeassistant/components/fritzhome.py

```diff
@@ -0,0 +1,58 @@
+"""
+Support for AVM Fritz!Box fritzhome devices.
+
+For more details about this component, please refer to the documentation at
+http://home-assistant.io/components/fritzhome/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.helpers import discovery
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pyfritzhome==0.3.5']
+
+SUPPORTED_DOMAINS = ['climate', 'switch']
+
+DOMAIN = 'fritzhome'
+
+DEFAULT_HOST = 'fritz.box'
+
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the fritzhome component."""
+    from pyfritzhome import Fritzhome, LoginError
+
+    conf = config[DOMAIN]
+    host = conf.get(CONF_HOST)
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
+
+    try:
+        fritz = Fritzhome(host=host, user=username, password=password)
+        fritz.login()
+        hass.data[DOMAIN] = fritz.get_devices()
```

Comment:
If you want to be able to reconnect from a platform, you should store the `fritz` instance in `hass.data[DOMAIN]` instead of the devices. Then you can both reconnect and get devices from the platforms.

Reply:
Pass the `fritz` instance to the device class at init, so you have access to that inside each entity.

Reply:
Each device has a reference to the fritz instance. So it is possible to do a reconnect with that.

Reply:
I would prefer if you store the `Fritzhome` instance in `hass.data` and then pass that to each entity, instead of accessing a protected attribute of the fritz device.

Reply:
changed according your comment